src/main/java/org/example/department/Department.java

```diff
@@ -0,0 +1,42 @@
+package org.example.department;
+
+import org.example.student.Student;
+
+import java.util.Set;
+
+public class Department {
+    private String name;
+    private Set<Student> students;
+
+    public Department(String name) {
+        this.name = name;
+    }
+
+    public Department(String name, Set<Student> students) {
```

Comment:
I think that if a Student object has a Department assigned, this should not contain a Set of students 
because the Department is something that can exist without any student.
That way it has a potential for some inconsistency -> you would add a Student to a department, but will forget to assign a Department for this student